### DISCUSSION TITLE
[UnoCSS Vs Code Extension]: Add Error for wrong duplicates.

### DISCUSSION THREAD

**Discussion by louiss0** [unocss/unocss#3081] [Ideas]
## Motivation
I decided not to use the issue because this is a design problem, not a bug. 
I have recently discovered that you can place two classes with the same prefix in a class name without any errors at all.
You are not supposed to do this. 

This code is allowed with no errors.

```html
<div class="mx-4 mx-6">
  Div with two margin classes same prefix different suffix
</div>

```

Two classes with the same prefix are not supposed to be used unless it's prefixed with a modifier. 
Then what should happen is that the extension should figure out whether you are using the same base class prefix without. 
The rest of the words being different.

Should not be allowed

```html
<div class="m-4 m-6"></div>
```

Should not be allowed

```html
<div class="hover:bg-gray-500 hover:bg-gray-400"></div>
```

Can be allowed.

```html
<div class="hover:text-gray-500 hover:bg-gray-400"></div>
```

**Comment by texastoland** [comment]
Does Tailwind prevent this? It seems pretty complex!

**Reply by louiss0** [reply]
It does it's where I got the idea

**Reply by texastoland** [reply]
IFL this belongs in issues. The only place I can imagine detecting it is when Uno noramlizes styles before generating CSS. The reason is any shortcut or attribute could also contribute a conflict.
